pkg/operator/ceohelpers/machine_api.go

```diff
@@ -0,0 +1,42 @@
+package ceohelpers
+
+import (
+	machinelistersv1beta1 "github.com/openshift/client-go/machine/listers/machine/v1beta1"
+
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/utils/pointer"
+)
+
+// IsMachineAPIFunctionalWrapper returns a function for determining if the Machine API is functional.
+// As of today Machine API is functional when we find Machines resources in the Running state.
+//
+// Note:
+//  since this function needs to take into account only master machine objects
+//  make sure machineInformer contain only filtered data
+//  otherwise it might be expensive to react to every node update in larger installations
+//
+//  this function requires masterMachineSelector as a safety net, just in case a caller won't provide a filtered informer
+func IsMachineAPIFunctionalWrapper(hasMasterMachineInformerSyncedFn func() bool, masterMachineLister machinelistersv1beta1.MachineLister, masterMachineSelector labels.Selector) func() (bool, error) {
+	return func() (bool, error) {
```

Comment:
I prefer having structs containing members with functions against a method receiver as opposed to returning functions anonymous functinos like this.

Reply:
I wanted to avoid creating a struct with just a single method. I don't expect it to expose more methods, thus I decided to create a simple function instead.

Reply:
> I wanted to avoid creating a struct with just a single method. I don't expect it to expose more methods, thus I decided to create a simple function instead.

I don't find it simple.  I could also accept a direct `IsMachineAPIFunctional(hasMasterMachineInformerSyncedFn func() bool, masterMachineLister machinelistersv1beta1.MachineLister, masterMachineSelector labels.Selector)`

Reply:
I think I finally chased all the indirection through here to find this hardcoded.  If its only used in that controller, set it in that controller.  A unit test can still set the member to whatever you want for unit testing.

Reply:
It will be used in multiple controllers

Reply:
okay, I turned `IsMachineAPIFunctionalWrapper` into a `MachineAPI` struct that exposes `IsFunctional` method. See if you like it.